### PR TITLE
initial draft of ed25519-precompile cip 

### PR DESCRIPTION
**Pull request by joe-bowman**
* initial draft of cip for ed25519 precompile

### CODE REVIEW

**Comment by joe-bowman** [comment]
Draft implementation to follow next week.

File: CIPs/cip-TBC.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+- public key: the 32-octet Ed25519 public key of the signer
+- signature: the 64-octet Ed25519 signature
+
+`ED25519VFY` returns as output 4 octets:
+
+- `0x00000000` if signature is valid
+- any non-zero value indicates a signature verification failure
+
+### Address
+The address of `ED25519VFY` is `0x9`.
```
Comment (prestwich) [inline CIPs/cip-TBC.md:57]: can you change this to "to be assigned"?

File: CIPs/cip-TBC.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+`ED25519VFY` takes as input 128 octets:
+
+- message: the 32-octet message that was signed
+- public key: the 32-octet Ed25519 public key of the signer
+- signature: the 64-octet Ed25519 signature
+
+`ED25519VFY` returns as output 4 octets:
+
+- `0x00000000` if signature is valid
+- any non-zero value indicates a signature verification failure
```
Comment (prestwich) [inline CIPs/cip-TBC.md:54]: Will there be an initially defined error code? Is there any way for this precompile to revert?

**Comment by prestwich** [comment]
I'm good to merge as draft as soon as @YazzyYaz assigns a CIP number 👍

File: CIPs/cip-TBC.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+---
+cip: TBC
```
Comment (YazzyYaz) [inline CIPs/cip-TBC.md:2]: please add the number 25

Comment (YazzyYaz) [reply CIPs/cip-TBC.md:2]: Also please modify the file number from TBC to 0025 

Thank you so much for contributing this CIP!

**Review by YazzyYaz — APPROVED** [review]

**Pull request merged**